### PR TITLE
Michael Wave Electric Boogaloo

### PR DESCRIPTION
**Pull request by alsoandanswer**
Readds Microwaves. Nuff' said.

I'm not smart enough to code the backend, so the microwave is mostly cosmetic for now.

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/items/weapons/circuitboards/machinery/miscboards.dm [hunk 1]
```diff
@@ -95,6 +95,17 @@
...
+/obj/item/circuitboard/microwave
+	name = T_BOARD("microwave")
+	desc = "The circuitboard for a microwave."
+	build_path = /obj/machinery/appliance/cooker/microwave
+	origin_tech = list(TECH_MAGNET = 2, TECH_ENGINEERING = 2)
+	board_type = "machine"
+	req_components = list(
+							"/obj/item/stock_parts/capacitor" = 3,
+							"/obj/item/stock_parts/micro_laser" = 1,
+							"/obj/item/stock_parts/matter_bin" = 2)
```
Comment (SleepyGemmy) [inline code/game/objects/items/weapons/circuitboards/machinery/miscboards.dm:107]:
```suggestion
	req_components = list(
		"/obj/item/stock_parts/capacitor" = 3,
		"/obj/item/stock_parts/micro_laser" = 1,
		"/obj/item/stock_parts/matter_bin" = 2
	)
```

File: code/modules/cooking/machinery/cooking_machines/microwave.dm [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+
+	component_types = list(
+			/obj/item/circuitboard/microwave = 1,
+			/obj/item/stock_parts/capacitor = 3,
+			/obj/item/stock_parts/micro_laser = 1,
+			/obj/item/stock_parts/matter_bin = 2
+		)
+	starts_with = list(
+		/obj/item/reagent_containers/cooking_container/microwave
+	)
```
Comment (SleepyGemmy) [inline code/modules/cooking/machinery/cooking_machines/microwave.dm:41]:
```suggestion
	component_types = list(
		/obj/item/circuitboard/microwave = 1,
		/obj/item/stock_parts/capacitor = 3,
		/obj/item/stock_parts/micro_laser = 1,
		/obj/item/stock_parts/matter_bin = 2
	)

	starts_with = list(
		/obj/item/reagent_containers/cooking_container/microwave
	)
```

File: code/modules/cooking/machinery/cooking_machines/microwave.dm [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+		to_chat(user, SPAN_WARNING("\The [name] is dirty! You'll need to clean it before using it."))
+	else
+		ui_interact(user)
+
+/obj/machinery/appliance/cooker/microwave/examine(var/mob/user)
+	..()
+	if(broken > 0)
+		to_chat(user, "It's broken!")
+	else if(dirty >= 100)
+		to_chat(user, "The insides are completely filthy!")
```
Comment (SleepyGemmy) [inline code/modules/cooking/machinery/cooking_machines/microwave.dm:212]:
```suggestion
		to_chat(user, "The inside is completely filthy!")
```

File: code/modules/cooking/machinery/cooking_machines/microwave.dm [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+		else
+			to_chat(user, SPAN_WARNING("It's broken!"))
+			return 1
+	else if(dirty >= 100) // The microwave is all dirty so can't be used!
+		if(istype(O, /obj/item/reagent_containers/spray/cleaner) || istype(O, /obj/item/soap) || istype(O, /obj/item/reagent_containers/glass/rag)) // If they're trying to clean it then let them
+			user.visible_message( \
+				SPAN_NOTICE("\The [user] starts to clean the microwave."), \
+				SPAN_NOTICE("You start to clean the microwave.") \
+			)
+			if (do_after(user,20/O.toolspeed))
```
Comment (SleepyGemmy) [inline code/modules/cooking/machinery/cooking_machines/microwave.dm:121]: `use_tool` proc?

File: code/modules/cooking/machinery/cooking_machines/microwave.dm [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+		failed = FALSE
+		cook_time = update_cook_time(round(recipe.time * 2))
+
+	start()
+
+/obj/machinery/appliance/cooker/microwave/proc/update_cook_time(var/ct = 200)
+	RefreshParts()
+	return (ct / cooking_power)
+
+/obj/machinery/appliance/cooker/microwave/process() // What you see here are the remains of proc/wzhzhzh, 2010 - 2019. RIP.
```
Comment (SleepyGemmy) [inline code/modules/cooking/machinery/cooking_machines/microwave.dm:303]:
```suggestion
/obj/machinery/appliance/cooker/microwave/process()
```

File: code/modules/cooking/recipes/recipes_baked.dm [hunk 6]
```diff
@@ -144,8 +144,8 @@
 	reagents = list(/singleton/reagent/water = 5, /singleton/reagent/alcohol/vodka = 5, /singleton/reagent/psilocybin = 5)
 	result = /obj/item/reagent_containers/food/snacks/spacylibertyduff
 
-/singleton/recipe/hotdiggitydonk //heated donk, in lieu of a microwave
-	appliance = OVEN
+/singleton/recipe/hotdiggitydonk // MY MICROWAVE IS BACK, WOMAN
```
Comment (SleepyGemmy) [inline code/modules/cooking/recipes/recipes_baked.dm:147]:
```suggestion
/singleton/recipe/hotdiggitydonk
```

File: code/modules/cooking/recipes/recipes_microwave.dm [hunk 7]
```diff
@@ -0,0 +1,8 @@
+/singleton/recipe/microchips
+	appliance = MICROWAVE
+	items = list(
+		/obj/item/reagent_containers/food/snacks/rawsticks
+	)
+	result = /obj/item/reagent_containers/food/snacks/microchips
+
+// This place was long and storied and is now a ghost town.
```
Comment (SleepyGemmy) [inline code/modules/cooking/recipes/recipes_microwave.dm:8]:
```suggestion
```

File: code/modules/cooking/recipes/recipes_pan.dm [hunk 8]
```diff
@@ -14,15 +14,15 @@
 		for (var/obj/item/reagent_containers/food/snacks/amanitajelly/being_cooked in .)
 			being_cooked.reagents.del_reagent(/singleton/reagent/toxin/amatoxin)
 
-// Ports from the microwave... yeah
+// Ports from the microwave... yeah  // YEAH RIGHT
```
Comment (SleepyGemmy) [inline code/modules/cooking/recipes/recipes_pan.dm:18]:
```suggestion
```

File: html/changelogs/wezzy_michaelwave2.yml [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - rscadd: "Re-adds Microwaves."
```
Comment (SleepyGemmy) [inline html/changelogs/wezzy_michaelwave2.yml:41]:
```suggestion
  - rscadd: "Re-adds the microwave."
```

File: maps/sccv_horizon/sccv_horizon-3_deck_3.dmm [hunk 10]
```diff
@@ -440,7 +440,7 @@
 /obj/effect/floor_decal/corner/red/diagonal,
 /obj/structure/table/reinforced/wood,
 /obj/item/storage/box/fancy/donut,
-/obj/item/paper/fluff/microwave,
```
Comment (SleepyGemmy) [inline maps/sccv_horizon/sccv_horizon-3_deck_3.dmm:443]: this should be removed from the code.

**Review by SleepyGemmy — DISMISSED** [review]

File: code/modules/cooking/machinery/cooking_machines/microwave.dm (hunk 5 above)
Comment (atteria) [reply code/modules/cooking/machinery/cooking_machines/microwave.dm:303]: Not even a memorial for the greatest proc of all time...

Comment (alsoandanswer) [reply code/modules/cooking/machinery/cooking_machines/microwave.dm:303]: I don't understand your vendetta against developer comments, Gem.

Don't waste your time requesting changes to remove them, I won't commit them.

**Comment by alsoandanswer** [comment]
!review

**Comment by alsoandanswer** [comment]
!review

**Review by ScheveningenZonn — APPROVED** [review]
wzhzhzhzh

**Pull request closed without merging**